bcipy/task/orchestrator/orchestrator.py

```diff
@@ -69,28 +77,71 @@ def __init__(
         self.visualize = visualize
         self.progress = 0
 
-        self.ready_to_execute = True
+        self.ready_to_execute = False
         self.logger.info("Session Orchestrator initialized successfully")
 
     def add_task(self, task: Type[Task]) -> None:
+        """Add a task to the orchestrator"""
         self.tasks.append(task)
         self.task_names.append(task.name)
+        self.ready_to_execute = True
 
     def add_tasks(self, tasks: List[Type[Task]]) -> None:
+        """Add a list of tasks to the orchestrator"""
         for task in tasks:
             self.add_task(task)
+        self.ready_to_execute = True
+
+    def set_next_phrase(self) -> None:
+        """Set the next phrase to be copied from the list of copy phrases loaded or the parameters directly.
+
+        If there are no more phrases to copy, the task text and spelled letters from parameters will be used.
+        """
+        if self.copyphrases:
+            if len(self.copyphrases) > 0:
+                text, index = self.copyphrases.pop(0)
+                self.next_phrase = text
+                self.starting_index = index
+            else:
+                self.next_phrase = self.parameters['task_text']
+        self.parameters['task_text'] = self.next_phrase
+        self.parameters['spelled_letters_count'] = self.starting_index
+
+    def initialize_copy_phrases(self) -> None:
+        """Load copy phrases from a json file or take the task text if no file is provided.
+
+        Expects a json file structured as follows:
+        {
+            "Phrases": [
+                [string, int],
+                [string, int],
+                ...
+            ]
+        }
+        """
+        # load copy phrases from json file or take the task text if no file is provided
+        if self.parameters.get('copy_phrases_location') != "":
```

Comment:
Empty strings are treated as False so this can be shortened to: 

`if self.parameters.get('copy_phrases_location'):`